src/tsHelpers.ts

```diff
@@ -13,8 +13,8 @@ export type IsUnknown<T, True, False = never> = unknown extends T
   : False
 
 export type IsEmptyObj<T, True, False = never> = T extends any
```

Comment:
This is what I'm most concerned with. I believe this should be good to find really empty objects (the one before also matched objects with only optional properties), but I'm not 100% sure I'm not missing something.

Reply:
Your intuition is right! As far as I know, that's the most reliable way to match an empty object. `{}` is very tricky, since [it doesn't really mean _empty object_](https://gist.github.com/OliverJAsh/381cd397008309c4a95d8f9bd31adcd7). 😉

Reply:
Thanks a lot, that gives me some assurance :)

Reply:
The empty object detection looks good to me too. `keyof T extends never` is the most reliable way to ensure that the type has no keys.